test/Connection/Http1ConnectionTest.php

```diff
@@ -194,6 +198,74 @@ public function testInactivityTimeout(): \Generator
         }
     }
 
+    public function testWritingRequestWithRelativeUriPathFails(): \Generator
+    {
+        [$client] = Socket\createPair();
+
+        $connection = new Http1Connection($client, 5000);
+        $uri = $this->createMock(UriInterface::class);
+        $uri->method('getScheme')->willReturn('http');
+        $uri->method('getAuthority')->willReturn('');
+        $uri->method('getUserInfo')->willReturn('');
+        $uri->method('getHost')->willReturn('localhost');
+        $uri->method('getQuery')->willReturn('');
+        $uri->method('getFragment')->willReturn('');
+        $uri->method('withScheme')->willReturnSelf();
+        $uri->method('withUserInfo')->willReturnSelf();
+        $uri->method('withHost')->willReturnSelf();
+        $uri->method('withPort')->willReturnSelf();
+        $uri->method('withQuery')->willReturnSelf();
+        $uri->method('withFragment')->willReturnSelf();
+        $uri->method('__toString')->willReturn('http://localhost/foo');
```

Comment:
Do we need a mock here or can we just use `league/uri`?

Reply:
We need mock because `league/uri` doesn't allow relative paths and I don't think it's a good idea to require-dev another implementation that does allow them (`laminas/laminas-diactoros`, for example).

Reply:
That's good to know, thanks! I wouldn't have a problem with requiring `laminas/laminas-diactoros` in `require-dev`.

Reply:
That's not a big problem, but they can decide to change this behavior in future - I think it's better not to rely on it. Anyway, it's just one small negative corner-case.